### PR TITLE
Add Agent Minio Credential for Sensor Agent

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Tracking issue
https://github.com/flyteorg/flyte/issues/3936
https://github.com/flyteorg/flytesnacks/pull/1180

## Describe your changes
Now we can use the sensor agent without setting the Minio credential.
You can test it with my new image.
```bash
flytectl demo start --image futureoutlier/flyte-sandbox:minio-credential
```
## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Screenshots
image
image
image

### CODE REVIEW

File: charts/flyteagent/templates/agent/deployment.yaml [hunk 1]
```diff
@@ -26,28 +26,44 @@ spec:
...
         name: flyteagent
-        volumeMounts: {{- include "agentSecret.volumeMount" . | nindent 8 }}
         ports:
         - containerPort: {{ .Values.ports.containerPort }}
           name: {{ .Values.ports.name }}
         resources: {{- toYaml .Values.resources | nindent 10 }}
+        env:
+        - name: FLYTE_AWS_ENDPOINT
+          valueFrom:
+            secretKeyRef:
```
Comment (pingsutw) [inline charts/flyteagent/templates/agent/deployment.yaml:36]: It's not necessary to use `secretKeyRef`, just use `value` here.

File: charts/flyteagent/templates/agent/deployment.yaml [hunk 2]
```diff
@@ -26,28 +26,44 @@ spec:
...
-      nodeSelector: {{ tpl (toYaml .) $ | nindent 8 }}
+      nodeSelector: {{ tpl (toYaml .) $ | nindent 10 }}
       {{- end }}
       {{- with .Values.affinity }}
-      affinity: {{ tpl (toYaml .) $ | nindent 8 }}
+      affinity: {{ tpl (toYaml .) $ | nindent 10 }}
       {{- end }}
       {{- with .Values.tolerations }}
-      tolerations: {{ tpl (toYaml .) $ | nindent 8 }}
+      tolerations: {{ tpl (toYaml .) $ | nindent 10 }}
```
Comment (pingsutw) [inline charts/flyteagent/templates/agent/deployment.yaml:68]: why change it to 10?

Comment (Future-Outlier) [reply charts/flyteagent/templates/agent/deployment.yaml:68]: I am still testing, will finish it today, thanks

File: charts/flyteagent/templates/agent/deployment.yaml (hunk 1 above)
Comment (Future-Outlier) [reply charts/flyteagent/templates/agent/deployment.yaml:36]: thanks!

File: charts/flyteagent/README.md [hunk 3]
```diff
@@ -9,8 +9,9 @@ A Helm chart for Flyte agent
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | additionalContainers | list | `[]` | Appends additional containers to the deployment spec. May include template values. |
+| additionalEnvs | list | `[{"name":"FLYTE_AWS_ENDPOINT","value":"http://flyte-sandbox-minio.flyte:9000"},{"name":"FLYTE_AWS_ACCESS_KEY_ID","value":"minio"},{"name":"FLYTE_AWS_SECRET_ACCESS_KEY","value":"miniostorage"}]` | Appends additional volumes to the deployment spec. May include template values. |
 | additionalVolumeMounts | list | `[]` | Appends additional volume mounts to the main container's spec. May include template values. |
-| additionalVolumes | list | `[]` | Appends additional volumes to the deployment spec. May include template values. |
```
Comment (pingsutw) [inline charts/flyteagent/README.md:13]: why remove this

Comment (Future-Outlier) [reply charts/flyteagent/README.md:13]: Accident, XD

**Review by pingsutw — APPROVED** [review]

**Pull request merged**